src/actions/app.js

```diff
@@ -416,3 +418,20 @@ export function updateBrowserConnectionStatus(
 ): Action {
   return { type: 'UPDATE_BROWSER_CONNECTION_STATUS', browserConnectionStatus };
 }
+
+export function toggleOpenCategory(
+  category: IndexIntoCategoryList
+): ThunkAction<void> {
+  return (dispatch, getState) => {
+    let openCategories = [...getOpenCategories(getState())];
+    if (openCategories.includes(category)) {
+      openCategories = openCategories.filter((c) => c !== category);
+    } else {
+      openCategories.push(category);
+    }
+    dispatch({
+      type: 'CHANGE_OPEN_CATEGORIES',
+      openCategories: openCategories,
+    });
```

Comment:
It would be a good idea to dispatch this inside a callback from `withHistoryReplaceStateSync` (this is in `src/app-logic/url-handling.js`):

This could look like this:
```js
withHistoryReplaceStateSync(() => dispatch({
  type: 'CHANGE_OPEN_CATEGORIES',
  openCategories: openCategories,
}));
```

If we don't do that, each time we open a category will push a new state to the history, and pressing the back button will just change the  category toggle. With this tool, instead, the new state will replace the current state, and pressing "back" will go to the previous "big" change, which is what we want, I believe.
(side note: currently pressing "back" once crashes the profiler UI, but we should fix this).

Hope this is clear :)

Reply:
Makes sense.